### PR TITLE
LL-96

### PR DESCRIPTION
**Pull request by vliu36**
# Description

Please include a summary of the change and its subsequent issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Subsequent # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

1. Get client id and secret from Vinson 
2. Use a .env file or replace id and secret variables with the respective client id and secret (DO NOT PUSH TO GITHUB)

Client:
1. ``` cd client ```
2. ``` npm start ```
3. Click login/register buttons and follow the prompts
4. Check server console

Server:
1. Run ``` src/routes/github.ts ```

# Checklist (when relevant):

- [ ] My code follows the style guidelines of this project

    [TypeScript](https://google.github.io/styleguide/tsguide.html)
[Python](https://peps.python.org/pep-0008/)
[Schema](https://learn.microsoft.com/en-us/previous-versions/windows/desktop/forefront-2010/ee652242(v=vs.100))
- [ ] I have included a reviewer on the pull request [*Replace this text with the reviewer*]
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: client/src/components/ThirdParty.tsx [hunk 1]
```diff
@@ -54,9 +54,19 @@ export const MicrosoftLogin = () => {
 };
 
 export const GitHubLogin = () => {
+  const CLIENT_ID = process.env.GITHUB_CLIENT_ID;
```
Comment (pogi7) [inline client/src/components/ThirdParty.tsx:57]: To get `environment variables` in React to work follow this guide:
https://stackoverflow.com/questions/49108136/importing-env-variable-react-front-end

File: server/src/routes/githubTest.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { Request, Response } from 'express';
+const express = require('express');
+const app = express();
+
```
Comment (pogi7) [inline server/src/routes/githubTest.ts:4]: import this into `server/server.ts` file

File: server/src/routes/githubTest.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+app.get('/github', (req: Request, res: Response) => {
```
Comment (pogi7) [inline server/src/routes/githubTest.ts:5]: Use the express.router module.  Look at `server/routes/routes.ts` page.  Call the path with an `axios` call in the client end

**Review by pogi7 — CHANGES_REQUESTED** [review]
I can't merge this until the pipeline passes:
https://github.com/pogi7/luminosityleds/actions/runs/3946000651/jobs/6753378078

**Review by erikajledesma — CHANGES_REQUESTED** [review]
@vliu36 Just tested your code, not sure why I get the following error on the console: 
image

**Comment by pogi7** [comment]
@vliu36 Please fix the merge conflicts.  Let me know if you have any questions.

**Review by Verus20 — APPROVED** [review]
LGTM!

**Review by pogi7 — APPROVED** [review]
Lgtm

**Pull request merged**